dialoguekit/core/dialogue.py

```diff
@@ -103,6 +110,23 @@ def add_utterance(self, utterance: Utterance) -> None:
             )
         self._utterances.append(utterance)
 
+    def add_utterance_feedback(
+        self, utterance_feedback: UtteranceFeedback
+    ) -> None:
+        """Adds user's feedback on utterance level.
+
+        The feedback previously added for a given utterance is overwritten.
+
+        Args:
+            utterance_feedback: User's feedback.
+        """
+        self._utterance_feedbacks = [
+            feedback
+            for feedback in self._utterance_feedbacks
+            if feedback.utterance_id != utterance_feedback.utterance_id
+        ]
+        self._utterance_feedbacks.append(utterance_feedback)
```

Comment:
This is not very efficient, nor elegant. Why not store feedback as a dict, indexed by utterance_id? Then it's enough to set `self._utterance_feedbacks[utterance_feedback.utterance_id] = utterance_feedback`.

Reply:
Done